internal/controller/csiaddons/persistentvolumeclaim_controller.go

```diff
@@ -245,44 +172,48 @@ func (r *PersistentVolumeClaimReconciler) checkDriverSupportReclaimsSpace(logger
 //   - Check if the schedule is present in the PVC annotations. If yes, use that.
 //   - Check if the schedule is present in the namespace annotations. If yes,
 //     use that.
-//   - If schedule is not present in namespace annotations, return ErrorScheduleNotFound.
-//   - If schedule is present in namespace annotations, check for reclaimSpace
-//     support by the driver.
-//   - If driver supports reclaimSpace, use the schedule from namespace.
-//   - If driver does not support reclaimSpace, return ErrScheduleNotFound.
-//     Depending on requeue value, it will throw ErrorConnNotFoundRequeueNeeded.
+//   - Check if the schedule is present in the storageclass annotations. If yes,
+//     use that.
 func (r *PersistentVolumeClaimReconciler) determineScheduleAndRequeue(
 	ctx context.Context,
 	logger *logr.Logger,
 	pvc *corev1.PersistentVolumeClaim,
 	driverName string,
+	annotation string,
 ) (string, error) {
 	annotations := pvc.GetAnnotations()
-	schedule, scheduleFound := getScheduleFromAnnotation(logger, annotations)
+	schedule, scheduleFound := getScheduleFromAnnotation(annotation, logger, annotations)
 	if scheduleFound {
 		return schedule, nil
 	}
 
 	// check for namespace schedule annotation.
-	// We cannot have a generic solution for all CSI drivers to get the driver
-	// name from PV and check if driver supports space reclamation or not and
-	// requeue the request if the driver is not registered in the connection
-	// pool. This can put the controller in a requeue loop. Hence we are
-	// reading the driver name from the namespace annotation and checking if
-	// the driver is registered in the connection pool and if not we are not
-	// requeuing the request.
 	ns := &corev1.Namespace{}
 	err := r.Client.Get(ctx, types.NamespacedName{Name: pvc.Namespace}, ns)
 	if err != nil {
 		logger.Error(err, "Failed to get Namespace", "Namespace", pvc.Namespace)
 		return "", err
 	}
-	schedule, scheduleFound = getScheduleFromAnnotation(logger, ns.Annotations)
+	schedule, scheduleFound = getScheduleFromAnnotation(annotation, logger, ns.Annotations)
 
 	// If the schedule is found, check whether driver supports the
 	// space reclamation using annotation on namespace and registered driver
 	// capability for decision on requeue.
 	if scheduleFound {
+		// If the check is for key rotation
+		// return the schedule directly
+		if annotation == krcJobScheduleTimeAnnotation {
+			return schedule, nil
+		}
```

Comment:
we need to have a driver support check right? we cannot just return from here, the flow should remain same for both key rotation and reclaimspace.

Reply:
The capability is already checked for in `getClientByNode` in `EncryptionKeyRotationJobController`. If no nodeclient is found, the reconcile ends there. May I know the benefit it brings to add a second check here as well?